apps/web/pages/api/auth/two-factor/totp/disable.ts

```diff
@@ -28,17 +29,19 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     return res.status(401).json({ message: "Not authenticated" });
   }
 
-  if (!user.password) {
+  if (!user.password && user.identityProvider === IdentityProvider.CAL) {
     return res.status(400).json({ error: ErrorCode.UserMissingPassword });
   }
 
   if (!user.twoFactorEnabled) {
     return res.json({ message: "Two factor disabled" });
   }
 
-  const isCorrectPassword = await verifyPassword(req.body.password, user.password);
-  if (!isCorrectPassword) {
-    return res.status(400).json({ error: ErrorCode.IncorrectPassword });
+  if (user.password && user.identityProvider === IdentityProvider.CAL) {
```

Comment:
When disabling an account with 2FA with a provider not CAL, no password will come so no need to fail when is not correct